### ISSUE TITLE
[Spike] AIM: Implement Support for AIM Events

### ISSUE THREAD

**Issue by kbford56** [newrelic/newrelic-java-agent#1708]
### Description
For AI Monitoring, we want to allow customers to be able to sample AI events separately from existing custom events. To do this, the agent needs to be able to separate out the AI/LLM events from other custom events as well as provide configuration to tune settings such as the pool size for these events.

~~WIP spec: https://source.datanerd.us/agents/agent-specs/pull/659~~

### Acceptance Criteria
AI Monitoring occurs using a new AI-specific event.

### Design Consideration/Limitations
Allows customers to fine-tune the monitoring pool size without impacting other agent features.

### Dependencies
N/A

### Additional context
[CDD](https://newrelic.atlassian.net/wiki/spaces/INST/pages/3175579674/APM+Agents+Common+Events+Implementation)

**Comment by jasonjkeller** [comment]
This work is out of scope for GA

**Comment by kford-newrelic** [comment]
Need to check in the next quarter whether this needs to be actioned or not.

**Comment by jasonjkeller** [comment]
This work is no longer tied to the Bedrock milestone. Current proposals are being discussed on this DACI: https://newrelic.atlassian.net/wiki/spaces/INST/pages/3175579674/APM+Agents+Common+Events+Implementation

**Comment by jasonjkeller** [comment]
I don't believe this work is ready to be taken on yet. I'll check in on the status this week.
